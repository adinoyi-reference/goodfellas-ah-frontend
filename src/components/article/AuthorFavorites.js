import React, { Fragment } from 'react';
import ProfileArticle from './Profile';

const ProfileArticleList = (props) => {
  const { articles } = props;
  return (
    <Fragment>
      {
        articles.map((article) => <ProfileArticle key={article.id} author={props.author} authorImage={props.authorImage} type="favoriteArticle" article={article.article} />)
      }
    </Fragment>);
};

export default ProfileArticleList;
