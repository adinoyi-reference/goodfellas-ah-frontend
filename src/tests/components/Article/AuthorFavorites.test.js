import React from 'react';
import { shallow } from 'enzyme';
import AuthorFavorites from '../../../components/article/AuthorFavorites';
import * as profileData from '../../mock/profileData';

const setup = () => {
  const props = {
    articles: profileData.favoritedArticles.data.articles,
    userFullName: 'Afeez Awoyemi',
  };

  const enzymeWrapper = shallow(<AuthorFavorites {...props} />);

  return {
    props,
    enzymeWrapper,
  };
};

describe('components', () => {
  describe('ProfileToolbar component', () => {
    it('should render as expected', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
