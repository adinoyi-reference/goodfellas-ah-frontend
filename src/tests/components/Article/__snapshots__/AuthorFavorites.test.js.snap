// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ProfileToolbar component should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileArticleList
    articles={
      Array [
        Object {
          "authorId": 1,
          "averageRating": null,
          "body": "This is the fine long body this is the fine long body  this is the fine long body  this is the fine long body  this is the fine long body ",
          "comments": Array [],
          "createdAt": "2018-11-09T15:08:00.560Z",
          "description": " this is the fine long body  this is the fine long body ",
          "favorited": null,
          "favoritesCount": null,
          "id": 2,
          "image": null,
          "reactions": Array [],
          "read_time": "1 minute",
          "slug": "this-is-a-fine-title-which-i-would-really-love-to-have",
          "tagList": null,
          "title": "This is a fine title which I would really love to have",
          "updatedAt": "2018-11-09T15:08:00.560Z",
        },
      ]
    }
    userFullName="Afeez Awoyemi"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <_default
          type="favoriteArticle"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "article": undefined,
          "author": undefined,
          "authorImage": undefined,
          "type": "favoriteArticle",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <_default
            type="favoriteArticle"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "article": undefined,
            "author": undefined,
            "authorImage": undefined,
            "type": "favoriteArticle",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
